### PR TITLE
Remove document.createEvent("touchevent")

### PR DESCRIPTION
**Pull request by LanWei22**
Since in DOM spec, createEvent(interface) is not recommended and suggests "Event constructors ought to be used instead." Also, Chrome, Edge, Firefox and Safari do not support document.createEvent("touchevent"), we should just delete it.
DOM spec: https://dom.spec.whatwg.org/#dom-document-createevent.

### CODE REVIEW

**Comment by LanWei22** [comment]
cynthia@ could you please take a look, thank you.

File: touch-events/historical.html [hunk 1]
```diff
@@ -32,15 +32,6 @@
                "Should not be supported on the instance");
 }, "TouchList::identifiedTouch");
 
-test(function() {
-  assert_false("initTouchEvent" in TouchEvent.prototype,
-               "Should not be supported on the prototype");
-
-  var touchEvent = document.createEvent("touchevent");
-  assert_false("initTouchEvent" in touchEvent,
-               "Should not be supported on the instance");
```
Comment (Ms2ger) [inline touch-events/historical.html:41]: Keep this test, but use the constructor to create `touchEvent` instead.

**Review by Ms2ger — CHANGES_REQUESTED** [review]
You want to remove TouchEvent from `dom/nodes/Document-createEvent.js` and add it to the not-supported list in `dom/nodes/Document-createEvent.https.html`.

File: touch-events/historical.html (hunk 1 above)
Comment (LanWei22) [reply touch-events/historical.html]: Done, thanks!

**Review by LanWei22 — COMMENTED** [review]
> You want to remove TouchEvent from `dom/nodes/Document-createEvent.js` and add it to the not-supported list in `dom/nodes/Document-createEvent.https.html`.

Done, thank you.

**Comment by annevk** [comment]
This is still exposed on mobile as it happens. Properly specifying this in the DOM Standard depends on https://github.com/w3c/touch-events/issues/64. I'm not sure how we would expose that boolean in tests though.

**Review by Ms2ger — APPROVED** [review]

**Comment by Ms2ger** [comment]
@annevk gah, didn't see your comment before merging. Lemme know if I should revert.

**Comment by annevk** [comment]
I think it's fine. Once that issue is fixed we'll need to have another look at this and for desktop these are the correct results.

**Pull request merged**